instrumentation/aws_lambda/lib/opentelemetry/instrumentation/aws_lambda/handler.rb

```diff
@@ -0,0 +1,203 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsLambda
+      AWS_TRIGGERS = ['aws:sqs', 'aws:s3', 'aws:sns', 'aws:dynamodb'].freeze
+
+      # Handler class that creates a span around the _HANDLER
+      class Handler
+        attr_reader :handler_method, :handler_class
+
+        # anytime when update the code in a Lambda function or change the functional configuration,
+        # the next invocation results in a cold start; therefore these instance variable will be up-to-date
+        def initialize
+          @flush_timeout    = ENV.fetch('OTEL_INSTRUMENTATION_AWS_LAMBDA_FLUSH_TIMEOUT', '30000').to_i
+          @original_handler = ENV['ORIG_HANDLER'] || ENV['_HANDLER'] || ''
+          @handler_class    = nil
+          @handler_method   = nil
+          @handler_file     = nil
+
+          resolve_original_handler
+        end
+
+        # We want to capture the error if user's handler is causing issue
+        # but our wrapper and handler shouldn't cause any issue
+        def call_wrapped(event:, context:)
+          parent_context = extract_parent_context(event)
+
+          span_kind = nil
+          span_kind = if event['Records'] && AWS_TRIGGERS.include?(event['Records'].dig(0, 'eventSource'))
+                        :consumer
+                      else
+                        :server
+                      end
+
+          original_handler_error = nil
+          original_response = nil
+          OpenTelemetry::Context.with_current(parent_context) do
+            span_attributes = otel_attributes(event, context)
+            span = tracer.start_span(
+              @original_handler,
+              attributes: span_attributes,
+              kind: span_kind
+            )
+
+            begin
+              response = call_original_handler(event: event, context: context)
+              status_code = response['statusCode'] || response[:statusCode] if response.is_a?(Hash)
+              span.set_attribute(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE, status_code) if status_code
+            rescue StandardError => e
+              original_handler_error = e
+            ensure
+              original_response = response
+            end
+          rescue StandardError => e
+            span&.record_exception(e)
+            span&.status = OpenTelemetry::Trace::Status.error("Unhandled exception of type: #{e.class}")
```

Comment:
If I'm reading correctly this is handling an exception from instrumentation. In this case it should just be logged, not recorded. The Python example has try/catch [very specific to the calling of the wrapped function](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/fdcbbddb6c753e5e9d494ba399a5b4bcab4afc3f/instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py#L368-L373) (to record), then [other instrumention operations which might raise](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/fdcbbddb6c753e5e9d494ba399a5b4bcab4afc3f/instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py#L397-L401) are always tried and logged.

Reply:
Thanks! Updated!